### PR TITLE
Testing CSI/CCM migration

### LINKED ISSUE #1454: Add E2E tests for the CCM/CSI migration feature
This issue assumes adding a CI job that would:

* Create a cluster with in-tree cloud provider
* Create some resources depending on CCM/CSI (e.g. volumes and LoadBalancers)
* Migrate to external CCM/CSI
* Ensure existing resources are properly reconciled and try to create some additional resources
* Tear down the setup including created resources

### PR DESCRIPTION
**Pull request by kron4eg**
**What type of PR is this?**
/kind cleanup

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1454 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Testing CSI/CCM migration
```

### CODE REVIEW

File: pkg/apis/kubeone/helpers.go [hunk 1]
```diff
@@ -294,77 +294,92 @@ func (p CloudProviderSpec) CloudProviderInTree() bool {
...
+	featureGates, err := c.csiMigrationFeatureGates(complete)
+	if err != nil {
+		return nil, "", err
 	}
 
-	return nil
+	return featureGates, marshalFeatureGates(featureGates), nil
 }
 
 func marshalFeatureGates(fgm map[string]bool) string {
```
Comment (kron4eg) [inline pkg/apis/kubeone/helpers.go:385]: changes from above in whole file is done in the name of having a single place where CSI featureGates are defined

File: pkg/apis/kubeone/helpers.go [hunk 2]
```diff
@@ -294,77 +294,92 @@ func (p CloudProviderSpec) CloudProviderInTree() bool {
...
 
-	if complete {
-		for _, u := range c.InTreePluginUnregisterFeatureGate() {
-			featureGates[u] = true
-		}
+func (c KubeOneCluster) csiMigrationFeatureGates(complete bool) (map[string]bool, error) {
+	lessThan21Constraint, err := semver.NewConstraint("< 1.21.0")
+	if err != nil {
+		return nil, err
 	}
```
Comment (xmudrii) [inline pkg/apis/kubeone/helpers.go:311]: Nit (non-blocking): KubeOne 1.5 doesn't support Kubernetes 1.20 so we can drop this.

File: testv2/tests.yml [hunk 3]
```diff
@@ -312,6 +312,42 @@
...
     - name: aws_defaults
 
+- scenario: csi_ccm_migration
+  initVersion: v1.21.14
+  infrastructures:
+    - name: aws_defaults 
+    - name: openstack_default
+    - name: azure_default
+    - name: vsphere_default
+    - name: vsphere_flatcar
```
Comment (xmudrii) [inline testv2/tests.yml:322]: We can probably drop 1.21 tests because 1.21 reached EOL a while ago.

File: pkg/apis/kubeone/helpers.go (hunk 2 above)
Comment (kron4eg) [reply pkg/apis/kubeone/helpers.go]: Done, removed that stuff

File: testv2/tests.yml (hunk 3 above)
Comment (kron4eg) [reply testv2/tests.yml]: Done, removed that stuff

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

:zap: :shipit:

**Pull request merged**